parsers/regex/regex_test.go

```diff
@@ -0,0 +1,168 @@
+package regex
+
+import (
+	"errors"
+	"reflect"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/parsers"
+)
+
+const (
+	commonLogFormatTimeLayout = "02/Jan/2006:15:04:05 -0700"
+	iso8601TimeLayout         = "2006-01-02T15:04:05-07:00"
+)
+
+type testLineMap struct {
+	lineRegex string
+	input     string
+	expected  map[string]interface{}
+}
+
+// Test cases for RegexLineParser.ParseLine
+var tlms = []testLineMap{
+	{
+		// Simple word parsing
+		lineRegex: `(?P<word1>\w+) (?P<word2>\w+) (?P<word3>\w+)`,
+		input:     `apple banana orange`,
+		expected: map[string]interface{}{
+			"word1": "apple",
+			"word2": "banana",
+			"word3": "orange",
+		},
+	},
+	{
+		// Simple time parsing
+		lineRegex: `(?P<Year>\d{4})-(?P<Month>\d{2})-(?P<Day>\d{2})`,
+		input:     `2017-01-30 1980-01-02`, // Ignore the second date
+		expected: map[string]interface{}{
+			"Year":  "2017",
+			"Month": "01",
+			"Day":   "30",
+		},
+	},
+	{
+		// Fields containing whitespace
+		lineRegex: `\[(?P<BracketedField>[0-9A-Za-z\s]+)\] (?P<UnbracketedField>[0-9A-Za-z]+)`,
+		input:     `[some value] unbracketed`,
+		expected: map[string]interface{}{
+			"BracketedField":   "some value",
+			"UnbracketedField": "unbracketed",
+		},
+	},
+	{
+		// Nested regex grouping
+		lineRegex: `(?P<outer>[^ ]* (?P<inner1>[^ ]*) (?P<inner2>[^ ]*))`,
+		input:     `foo bar baz`,
+		expected: map[string]interface{}{
+			"outer":  "foo bar baz",
+			"inner1": "bar",
+			"inner2": "baz",
+		},
+	},
+	{
+		// Sample nginx log line
+		lineRegex: `(?P<time>\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}) \[(?P<status>.*)\].* request: "(?P<request>[^"]*)"`,
+		input:     `2017/11/07 22:59:46 [error] 5812#0: *777536449 connect() failed (111: Connection refused) while connecting to upstream, client: 127.0.0.1, server: localhost, request: "GET /isbns HTTP/1.1", upstream: "http://127.0.0.1:8080/isbns", host: "localhost"`,
+		expected: map[string]interface{}{
+			"time":    "2017/11/07 22:59:46",
+			"status":  "error",
+			"request": "GET /isbns HTTP/1.1",
+		},
+	},
+}
+
+func TestParseLine(t *testing.T) {
+	p := &Parser{
```

Comment:
It might be nice to call `p.Init(Options{...})` instead of populating the struct members directly, in order to also test that `Init()` does the right thing. (And maybe also worth adding a test to check that `Init()` fails sensibly if you give it a bad regex)

Reply:
weird, none of the other parser test files appear to test for `p.Init(...)` success

I'll fix it in this file and add a note to look thru the rest of honeytail in a separate diff :)